sys/src/9/pc/trap.c

```diff
@@ -616,6 +616,13 @@ extern void rdmsrfail(void);
 static void
 faultgpf(Ureg* ureg, void*)
 {
+	int user = (ureg->cs & 0xFFFF) == UESEL;
```

Comment:
There's a macro in fns.h
fns.h:#define   userureg(ur) (((ur)->cs & 0xFFFF) == UESEL)
might want to use that.

Reply:
Sure. Fixed up a few other locations of that sequence while I was at it; I left one (which is in a large expression also looking for data segments).